adapters/repos/db/lsmkv/segment_replace_strategy.go

```diff
@@ -91,10 +91,15 @@ func (s *segment) getBySecondary(pos int, key []byte) ([]byte, error) {
 	// invalid memory without the copy, thus leading to a SEGFAULT.
 	// Similar approach was used to fix SEGFAULT in collection strategy
 	// https://github.com/weaviate/weaviate/issues/1837
-	contentsCopy := make([]byte, node.End-node.Start)
+	var contentsCopy []byte
+	if uint64(len(buffer)) >= node.End-node.Start {
+		contentsCopy = buffer
```

Comment:
shouldn't be `buffer` trimmed to the required size (`contentsCopy = buffer[:node.End-node.Start]`) not to mix just read data with some garbage from previous reads at the end (or whatever data was stored in the buffer before)?

(I know in that case it is safe as `s.replaceStratParseData(contentsCopy)` reads exactly as much as it is supposed to do; that is rather generic remark, as same approach is used in `VectorFromBinary` method)

Instead of checking `len(buffer)`, `cap(buffer)` can be used, and buffer extended if needed up to its capacity.
Something like:
```
if uint64(cap(buffer)) >= node.End-node.Start {
	contentsCopy = buffer[:node.End-node.Start]
```